### ISSUE TITLE
Setup GitHub Continuous Integration

### ISSUE THREAD

**Issue by mike-moore** [super-owesome/bobble_controllers#11]
Would like to get CI going in our GitHub repository. This should be a good reference https://github.com/ros-industrial/industrial_ci

**Comment by dwes7** [comment]
I started looking into this today. I assumed you'd want to use Github actions since they're free and the industrial_ci has some scripts to help set it up. So I forked the bobble_controllers repo and tried putting in the [script](https://github.com/ros-industrial/industrial_ci#for-github-actions) from the industrial_ci quickstart guide for github actions and pushing it to my forked repo. The action steps seemed to work correctly, but I got the following result from the action. See the log file attached or see build result [here](https://github.com/dwes7/bobble_controllers/runs/1288228600?check_suite_focus=true).
[logs_2.zip](https://github.com/super-owesome/bobble_controllers/files/5417807/logs_2.zip)

It looks everything was good until the tests were ran. I haven't tried running the tests locally to see if they pass, or if its has something to do with the way the environment is setup in the github action.

**Comment by dwes7** [comment]
I also noticed that there's a [.gitlab-ci.yml](https://github.com/dwes7/bobble_controllers/blob/master/.gitlab-ci.yml), already in place. Is there already some CI for gitlab? Should I be building off of this script instead of focusing on github Actions? Or do we want to migrate over to github Actions instead? I guess Github actions didn't get released until Nov 2019, you pushed the .gitlab-ci.yml in April 2019 aedf995642ae6a439283002b002dd129dddd6a4b, so maybe sticking with the github ecosystem seems to make more sense?

**Comment by mike-moore** [comment]
@dwes7 yeah the idea would be to build off what's called out in .gitlab-ci. We would want to replicate that behavior using what GitHub provides.
